tools/mo/openvino/tools/mo/moc_frontend/pytorch_frontend_utils.py

```diff
@@ -23,16 +24,68 @@ def remove_tmp_onnx_model(out_dir):
             os.remove(tmp_onnx_model)
 
 
-def get_pytorch_decoder(model, input_shape, example_inputs, input_info):
+
+def get_pytorch_decoder(model, input_shape, example_inputs, args):
     try:
         from openvino.frontend.pytorch.decoder import TorchScriptPythonDecoder
     except Exception as e:
         log.error("PyTorch frontend loading failed")
         raise e
-    inputs = prepare_torch_inputs(example_inputs, input_shape, input_info, allow_none=True)
+    inputs = prepare_torch_inputs(example_inputs, input_shape, args.get("input"), allow_none=True)
     decoder = TorchScriptPythonDecoder(model, example_input=inputs)
+    args['input_model'] = decoder
+    args["framework"] = "pytorch"
+    if inputs is not None:
+        extract_input_info_from_example(inputs, args)
         
-    return decoder
+    return args
+
+
+def extract_input_info_from_example(example_inputs, args):
+    import openvino
```

Comment:
Please remove it and properly import routine for `InputCutInfo`

Reply:
unfortunatly, I can not import it at the top level due to circular dependecies